generator/wrapper_gen.py

```diff
@@ -175,7 +175,7 @@ def my_procid():
 
 def _get_data(x):
     if isinstance(x, pdDataFrame):
-        x = [x.loc[:,i].values for i in x]
+        x = x.values
```

Comment:
Here we should understand the memory layout of DataFrame.
We need to determine two significant for us cases appropriate to Homogen and SOA Numeric Tables and try to to except unnecessary copying of data.
Also we should pay attention to data types of columns. They can be as mix of ints and floats with various sizes but "aligned".
This is my current representation of it:
```
def _get_data(x):
    if isinstance(x, pdDataFrame):
        # get first row of Dataframe
        firstRow = df.iloc[0].values
        # try to detect c-contiguous representation of DF in memory
        if (firstRow.flags["C_CONTIGUOUS"] == True
            and firstRow.flags["OWNDATA"] == False):
            return x.values
        # np.ascontiguousarray will create a copy only when it is necessary
        # and will return the same memory otherwise
        x = [np.ascontiguousarray(x.iloc[:,i].values) for i in range(len(x.columns))]
    elif isinstance(x, pdSeries):
        x = [x.values]
    return x
```
This is a poor try to understand is the memory layout of DataFrame the c-contiguous array or not, it should be modified.

Reply:
This looks better, but is not what this PR does :)

If a series is not contiguous, I would expect that all series of the DF are non-contiguous. Is this a fair assumption to make? If all series' are non-contiguous, you probably better just use x.values.

You also have to make the elif case contiguous.

Another solution would be to implement a NT which supports non-contiguous SOA. Basically a variant of daal4py's NpyTable. I'd rather go that route than implicitly copying. We could have a way to show performance warnings and let the user copy to contiguous explicitly.